kevm_pyk/solc_to_k.py

```diff
@@ -0,0 +1,289 @@
+    symbolTable['EVMC_UNDEFINED_INSTRUCTION_NETWORK_ExceptionalStatusCode']  = lambda: 'EVMC_UNDEFINED_INSTRUCTION'
+    symbolTable['EVMC_SUCCESS_NETWORK_EndStatusCode']                        = lambda: 'EVMC_SUCCESS'
+    symbolTable['EVMC_STATIC_MODE_VIOLATION_NETWORK_ExceptionalStatusCode']  = lambda: 'EVMC_STATIC_MODE_VIOLATION'
+    symbolTable['EVMC_STACK_UNDERFLOW_NETWORK_ExceptionalStatusCode']        = lambda: 'EVMC_STACK_UNDERFLOW'
+    symbolTable['EVMC_STACK_OVERFLOW_NETWORK_ExceptionalStatusCode']         = lambda: 'EVMC_STACK_OVERFLOW'
+    symbolTable['EVMC_REVERT_NETWORK_EndStatusCode']                         = lambda: 'EVMC_REVERT'
+    symbolTable['EVMC_REJECTED_NETWORK_StatusCode']                          = lambda: 'EVMC_REJECTED'
+    symbolTable['EVMC_PRECOMPILE_FAILURE_NETWORK_ExceptionalStatusCode']     = lambda: 'EVMC_PRECOMPILE_FAILURE'
+    symbolTable['EVMC_OUT_OF_GAS_NETWORK_ExceptionalStatusCode']             = lambda: 'EVMC_OUT_OF_GAS'
+    symbolTable['EVMC_INVALID_MEMORY_ACCESS_NETWORK_ExceptionalStatusCode']  = lambda: 'EVMC_INVALID_MEMORY_ACCESS'
+    symbolTable['EVMC_INVALID_INSTRUCTION_NETWORK_ExceptionalStatusCode']    = lambda: 'EVMC_INVALID_INSTRUCTION'
+    symbolTable['EVMC_INTERNAL_ERROR_NETWORK_StatusCode']                    = lambda: 'EVMC_INTERNAL_ERROR'
+    symbolTable['EVMC_FAILURE_NETWORK_ExceptionalStatusCode']                = lambda: 'EVMC_FAILURE'
+    symbolTable['EVMC_CALL_DEPTH_EXCEEDED_NETWORK_ExceptionalStatusCode']    = lambda: 'EVMC_CALL_DEPTH_EXCEEDED'
+    symbolTable['EVMC_BALANCE_UNDERFLOW_NETWORK_ExceptionalStatusCode']      = lambda: 'EVMC_BALANCE_UNDERFLOW'
+    symbolTable['EVMC_BAD_JUMP_DESTINATION_NETWORK_ExceptionalStatusCode']   = lambda: 'EVMC_BAD_JUMP_DESTINATION'
+    symbolTable['EVMC_ACCOUNT_ALREADY_EXISTS_NETWORK_ExceptionalStatusCode'] = lambda: 'EVMC_ACCOUNT_ALREADY_EXISTS'
+    return symbolTable
+
+
+# Helpers
+
+def _extract_storage_sentences(contract_name, storage_sort, storage_layout):
+    types = storage_layout['types']
+
+    def recur(syntax, lhs, rhs, var_idx, type_name):
+        type_dict = types[type_name]
+        encoding = type_dict['encoding']
+
+        if encoding == 'inplace':
+            members = type_dict.get('members')
+            if members:
+                return recur_struct(syntax, lhs, rhs, var_idx, members)
+
+            type_label = type_dict['label']
+            return recur_value(syntax, lhs, rhs, type_label)
+
+        if encoding == 'mapping':
+            key_type_name = type_dict['key']
+            value_type_name = type_dict['value']
+            return recur_mapping(syntax, lhs, rhs, var_idx, key_type_name, value_type_name)
+
+        raise ValueError(f'Unsupported encoding: {encoding}')
+
+    def recur_value(syntax, lhs, rhs, type_label):
+        if type_label not in {'address', 'uint256', 'bytes32'}:
```

Comment:
This same check exists in multiple places. Either it should be factored out so we have a single common list of supported types or the code sholud be refactored.

Reply:
Also, `make tests/specs/examples/erc20-spec.md.prove` fails because it requires a `uint8`, which is not in this supported list.

Reply:
This prototype version does not support packing of multiple items into a single storage slot, thus for safety it raises an exception for small types (to be precise, for types that do not take more than half of a slot) like `uint8`.

What is your preference: should this feature be implemented in this PR, or should we merge first?

Reply:
Well we'll need it implemented, because `tests/specs/examples/erc20-spec.md.prove` currently works and currently uses it. But we could also have a flag `--no-storage-slots`, which keeps the python script from emitting the storage slots, and pass that flag for that test.

Reply:
I think the `--no-storage-slots` feature is valuable in itself, and is also relatively easy to implement, so I'll start with that to unblock merging this PR.

Created an issue for the rest: https://github.com/runtimeverification/erc20-verification/issues/279